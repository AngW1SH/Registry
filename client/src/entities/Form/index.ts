import type { IForm } from "./types/types";
import FormCard from "./ui/FormCard";
import { staticForms } from "./static/staticForms";

export type { IForm };
export { FormCard, staticForms };
