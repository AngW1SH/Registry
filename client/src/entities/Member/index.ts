import type { IMember } from "./types/types";
import { getMembersByMemberIds } from "./model/getMembersByMemberIds";
import { staticMembers } from "./static/staticMembers";

export type { IMember };
export { getMembersByMemberIds, staticMembers };
