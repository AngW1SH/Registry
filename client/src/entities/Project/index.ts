import ProjectCard from "./ui/ProjectCard";
import ProjectCardAlt from "./ui/ProjectCardAlt";
import ProjectCardWithStatus from "./ui/ProjectCardWithStatus";
import { staticProjects } from "./static/staticProjects";
import { getProjectStage } from "./utils/getProjectStage";
import {
  getProjectSingleFromDTO,
  getProjectFromDTO,
} from "./utils/getProjectFromDTO/getProjectFromDTO";
import type {
  ProjectDTO,
  IProject,
  IProjectSingle,
  IProjectSingleDTO,
} from "./types/types";
import { ProjectStage } from "./types/types";

export {
  ProjectCard,
  ProjectCardAlt,
  ProjectCardWithStatus,
  staticProjects,
  getProjectStage,
  ProjectStage,
  getProjectFromDTO,
  getProjectSingleFromDTO,
};
export type { ProjectDTO, IProject, IProjectSingle, IProjectSingleDTO };
