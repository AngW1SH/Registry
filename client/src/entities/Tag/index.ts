import TagSlider from "./ui/TagSlider";
import TagList from "./ui/TagList";

export { TagSlider, TagList };
