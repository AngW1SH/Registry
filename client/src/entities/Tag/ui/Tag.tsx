import { ButtonTransparentAlt } from "@/shared/ui";
import { FC, ReactNode } from "react";

interface TagProps {
  children: ReactNode;
  className?: string;
  selected?: boolean;
}

const Tag: FC<TagProps> = ({ children, className = "", selected = false }) => {
  if (selected)
    return (
      <ButtonTransparentAlt
        className={"w-max bg-black px-5 py-2 text-white " + className}
      >
        {children}
      </ButtonTransparentAlt>
    );

  return (
    <ButtonTransparentAlt className={"w-max px-5 py-2 " + className}>
      {children}
    </ButtonTransparentAlt>
  );
};

export default Tag;
