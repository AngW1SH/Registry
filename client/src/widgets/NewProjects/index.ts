import NewProjects from "./ui/NewProjects";

export { NewProjects };
