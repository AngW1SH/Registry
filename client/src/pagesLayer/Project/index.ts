import ProjectPage from "./ui/ProjectPage";
import { generateProjectPageMetadata } from "./ui/ProjectPage";

export { ProjectPage, generateProjectPageMetadata };
