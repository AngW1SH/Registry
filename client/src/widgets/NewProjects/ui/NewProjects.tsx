import { IProject, ProjectCardAlt } from "@/entities/Project";
import { ITag, TagList, getTagsByTagIds, staticTags } from "@/entities/Tag";
import { LinkWithIcon } from "@/shared/ui";
import { FC } from "react";
import { staticProjects } from "../static/staticProjects";
import { fetchNewProjects } from "../api/fetchNewProjects";

interface NewProjectsProps {
  data?: {
    projects: IProject[];
    tags: ITag[];
  };
}

const NewProjects: FC<NewProjectsProps> = async ({ data }) => {
  const { projects, tags } = data ? data : await fetchNewProjects();
  return (
    <>
      <ul className="grid auto-rows-fr grid-cols-1 gap-y-5 sm:grid-cols-2 lg:grid-cols-3 sm:[&>li:nth-child(2n)]:border-r-0 lg:[&>li:nth-child(2n)]:border-r lg:[&>li:nth-child(3n)]:border-r-0 [&>li]:border-x-0  sm:[&>li]:border-r">
        {projects.map((project) => (
          <li
            key={project.id}
            className="col-span-1 row-span-1 border-t border-black p-3"
          >
            <ProjectCardAlt
              project={project}
              tags={<TagList tags={getTagsByTagIds(project.tags, tags)} />}
            />
          </li>
        ))}
      </ul>
      <div className="pt-6" />
      <LinkWithIcon href="/" className="font-medium" iconSize={10}>
        Показать ещё
      </LinkWithIcon>
    </>
  );
};

export default NewProjects;
