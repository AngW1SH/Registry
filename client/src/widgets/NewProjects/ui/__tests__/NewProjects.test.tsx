import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import NewProjects from "../NewProjects";
import { staticProjects } from "@/entities/Project";
import { staticTags } from "@/entities/Tag";

describe("NewProjects Widget UI", () => {
  it("renders projects", async () => {
    const Component = NewProjects({
      data: { projects: staticProjects, tags: staticTags },
    });
    // @ts-ignore
    const { getByText } = render(await Component);

    expect(getByText(staticProjects[0].name)).toBeInTheDocument();
  });

  test("snapshot test", async () => {
    const Component = NewProjects({
      data: { projects: staticProjects, tags: staticTags },
    });
    // @ts-ignore
    const { asFragment } = render(await Component);

    expect(asFragment()).toMatchSnapshot();
  });
});
