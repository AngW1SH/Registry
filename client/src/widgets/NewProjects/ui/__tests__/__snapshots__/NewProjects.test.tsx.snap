// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewProjects Widget UI snapshot test 1`] = `
<DocumentFragment>
  <ul
    class="grid auto-rows-fr grid-cols-1 gap-y-5 sm:grid-cols-2 lg:grid-cols-3 sm:[&>li:nth-child(2n)]:border-r-0 lg:[&>li:nth-child(2n)]:border-r lg:[&>li:nth-child(3n)]:border-r-0 [&>li]:border-x-0  sm:[&>li]:border-r"
  >
    <li
      class="col-span-1 row-span-1 border-t border-black p-3"
    >
      <div
        class="flex flex-col "
      >
        <div
          class="flex justify-between"
        >
          <div
            class="w-15 whitespace-nowrap text-xs font-bold"
          >
            <div
              class="pt-1"
            />
            <span
              class="block text-3xl"
            >
              01
            </span>
            сен 2023
          </div>
          <div
            class="pr-7"
          />
          <div>
            <h2
              class="text-sm font-medium"
            >
              Биология растений в эпоху глобальных изменений климата
            </h2>
          </div>
        </div>
        <div
          class="ml-auto sm:ml-0 xl:pl-20"
        >
          <div
            class="pt-5 xl:pt-10"
          />
          <div>
            <ul
              class="flex flex-wrap gap-1 undefined"
            >
              <li>
                <button
                  class="cursor-pointer rounded-3xl border-2 border-black px-3 py-2 font-bold text-black w-max border border-[#9a9a9a] px-5 py-2 text-sm font-normal text-[#303030] "
                  type="button"
                >
                  Генетика
                </button>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </li>
    <li
      class="col-span-1 row-span-1 border-t border-black p-3"
    >
      <div
        class="flex flex-col "
      >
        <div
          class="flex justify-between"
        >
          <div
            class="w-15 whitespace-nowrap text-xs font-bold"
          >
            <div
              class="pt-1"
            />
            <span
              class="block text-3xl"
            >
              01
            </span>
            фев 2023
          </div>
          <div
            class="pr-7"
          />
          <div>
            <h2
              class="text-sm font-medium"
            >
              ГМО: история, достижения,социальные и экологические риски
            </h2>
          </div>
        </div>
        <div
          class="ml-auto sm:ml-0 xl:pl-20"
        >
          <div
            class="pt-5 xl:pt-10"
          />
          <div>
            <ul
              class="flex flex-wrap gap-1 undefined"
            >
              <li>
                <button
                  class="cursor-pointer rounded-3xl border-2 border-black px-3 py-2 font-bold text-black w-max border border-[#9a9a9a] px-5 py-2 text-sm font-normal text-[#303030] "
                  type="button"
                >
                  Медицина
                </button>
              </li>
              <li>
                <button
                  class="cursor-pointer rounded-3xl border-2 border-black px-3 py-2 font-bold text-black w-max border border-[#9a9a9a] px-5 py-2 text-sm font-normal text-[#303030] "
                  type="button"
                >
                  Генетика
                </button>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </li>
    <li
      class="col-span-1 row-span-1 border-t border-black p-3"
    >
      <div
        class="flex flex-col "
      >
        <div
          class="flex justify-between"
        >
          <div
            class="w-15 whitespace-nowrap text-xs font-bold"
          >
            <div
              class="pt-1"
            />
            <span
              class="block text-3xl"
            >
              09
            </span>
            сен 2023
          </div>
          <div
            class="pr-7"
          />
          <div>
            <h2
              class="text-sm font-medium"
            >
              Современные проблемы эпидемиологии, микробиологии и гигиены
            </h2>
          </div>
        </div>
        <div
          class="ml-auto sm:ml-0 xl:pl-20"
        >
          <div
            class="pt-5 xl:pt-10"
          />
          <div>
            <ul
              class="flex flex-wrap gap-1 undefined"
            >
              <li>
                <button
                  class="cursor-pointer rounded-3xl border-2 border-black px-3 py-2 font-bold text-black w-max border border-[#9a9a9a] px-5 py-2 text-sm font-normal text-[#303030] "
                  type="button"
                >
                  Общество
                </button>
              </li>
              <li>
                <button
                  class="cursor-pointer rounded-3xl border-2 border-black px-3 py-2 font-bold text-black w-max border border-[#9a9a9a] px-5 py-2 text-sm font-normal text-[#303030] "
                  type="button"
                >
                  Социология
                </button>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </li>
    <li
      class="col-span-1 row-span-1 border-t border-black p-3"
    >
      <div
        class="flex flex-col "
      >
        <div
          class="flex justify-between"
        >
          <div
            class="w-15 whitespace-nowrap text-xs font-bold"
          >
            <div
              class="pt-1"
            />
            <span
              class="block text-3xl"
            >
              07
            </span>
            сен 2023
          </div>
          <div
            class="pr-7"
          />
          <div>
            <h2
              class="text-sm font-medium"
            >
              Изучение социально-экономических проблем соверменного испанского общества
            </h2>
          </div>
        </div>
        <div
          class="ml-auto sm:ml-0 xl:pl-20"
        >
          <div
            class="pt-5 xl:pt-10"
          />
          <div>
            <ul
              class="flex flex-wrap gap-1 undefined"
            >
              <li>
                <button
                  class="cursor-pointer rounded-3xl border-2 border-black px-3 py-2 font-bold text-black w-max border border-[#9a9a9a] px-5 py-2 text-sm font-normal text-[#303030] "
                  type="button"
                >
                  Общество
                </button>
              </li>
              <li>
                <button
                  class="cursor-pointer rounded-3xl border-2 border-black px-3 py-2 font-bold text-black w-max border border-[#9a9a9a] px-5 py-2 text-sm font-normal text-[#303030] "
                  type="button"
                >
                  Социология
                </button>
              </li>
              <li>
                <button
                  class="cursor-pointer rounded-3xl border-2 border-black px-3 py-2 font-bold text-black w-max border border-[#9a9a9a] px-5 py-2 text-sm font-normal text-[#303030] "
                  type="button"
                >
                  Финансы
                </button>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </li>
  </ul>
  <div
    class="pt-6"
  />
  <a
    href="/"
  >
    <button
      class="relative font-medium"
      style="padding-right: 20px;"
    >
      Показать ещё
      <div
        class="absolute right-0 top-[calc(50%-5px)]"
        style="height: 10px; width: 10px; top: calc(50% - 5px);"
      >
        <img
          alt=""
          class="object-contain"
          data-nimg="fill"
          decoding="async"
          loading="lazy"
          src="/link-arrow-black.svg"
          style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
        />
      </div>
    </button>
  </a>
</DocumentFragment>
`;
